qcodes/dataset/data_set.py

```diff
@@ -205,6 +206,36 @@ def _clean_up(self) -> None:
         self.log.debug("Stopped subscriber")
 
 
+class _BackgroundWriter(Thread):
+    """
+    Write the results from the DataSet's dataqueue in a new thread
+    """
+
+    def __init__(self, queue: Queue, conn: ConnectionPlus, table_name: str):
+        super().__init__()
+        self.queue = queue
+        self.path = path_to_dbfile(conn)
+        self.table_name = table_name
+        self.keep_writing = True
+
+    def run(self) -> None:
+
+        self.conn = connect(self.path)
+
+        while self.keep_writing:
+
+            item = self.queue.get()
+            if item['keys'] == 'stop':
+                self.keep_writing = False
+                self.conn.close()
+            else:
+                self.write_results(item['keys'], item['values'])
+            self.queue.task_done()
+
+    def write_results(self, keys: List[str],
+                      values: List[List[Any]]) -> None:
```

Comment:
I suggest using Sequence as the type for input arguments to make it clear that they are not modified

Reply:
The signature of `write_results` here is made to match the signature of `insert_many_values` (source: https://github.com/QCoDeS/Qcodes/blob/61eb02f460a3777fa7d247d85201cab78970cabb/qcodes/dataset/sqlite/query_helpers.py#L151). Should we change both?

Reply:
Yes, I think those are from times where we knew less of mypy

Reply:
Done.